implementation/servlet/src/main/java/net/opentsdb/servlet/resources/AsyncRunner.java

```diff
@@ -0,0 +1,147 @@
+// This file is part of OpenTSDB.
+// Copyright (C) 2020 The OpenTSDB Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package net.opentsdb.servlet.resources;
+
+import java.io.IOException;
+
+import javax.servlet.AsyncContext;
+import javax.servlet.AsyncEvent;
+import javax.servlet.AsyncListener;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.opentsdb.exceptions.QueryExecutionException;
+import net.opentsdb.query.QueryContext;
+import net.opentsdb.query.QueryMode;
+import net.opentsdb.servlet.exceptions.GenericExceptionMapper;
+import net.opentsdb.stats.Span;
+import net.opentsdb.stats.Trace;
+
+/**
+ * Static helper to handle an async query. 
+ * 
+ * @since 3.0
+ */
+public class AsyncRunner {
+  private static final Logger LOG = LoggerFactory.getLogger(AsyncRunner.class);
+  
+  private AsyncRunner() { }
```

Comment:
Do you need this?

Reply:
Yeah cause I don't want anyone instantiating it. It's how you mark a static class as intending to hold static methods only..